### ISSUE TITLE
RuntimeError: probability tensor contains either `inf`, `nan` or element < 0 [Usage] 

### ISSUE THREAD

**Issue by wanghao-cst** [haotian-liu/LLaVA#450]
### RuntimeError: probability tensor contains either inf, nan or element < 0

Issue:
When I use the lora weights, both llama2 and vicuna7b got the same error:
Command:
```
python -m llava.eval.run_llava \
    --model-path "/home/wanghao/weights/llava/llava-llama-2-7b-chat-lightning-lora-preview" \
    --model-base "/home/wanghao/weights/llava/llama-2-7b-chat-hf" \
    --image-file "llava/serve/examples/extreme_ironing.jpg" \
    --query "What is unusual about this image?"
```

I have made sure that the ckpt are correct, like the base model should in hf format.

Screenshots:
<img width="1163" alt="截屏2023-09-21 20 42 17" src="https://github.com/haotian-liu/LLaVA/assets/55015183/c9731dbe-1572-4857-8931-19159b9b4280">

**Comment by 478786359** [comment]
> ### RuntimeError: probability tensor contains either inf, nan or element < 0
> Issue: When I use the lora weights, both llama2 and vicuna7b got the same error: Command:
> 
> ```
> python -m llava.eval.run_llava \
>     --model-path "/home/wanghao/weights/llava/llava-llama-2-7b-chat-lightning-lora-preview" \
>     --model-base "/home/wanghao/weights/llava/llama-2-7b-chat-hf" \
>     --image-file "llava/serve/examples/extreme_ironing.jpg" \
>     --query "What is unusual about this image?"
> ```
> 
> I have made sure that the ckpt are correct, like the base model should in hf format.
> 
> Screenshots: <img alt="截屏2023-09-21 20 42 17" width="1163" src="https://user-images.githubusercontent.com/55015183/269615330-c9731dbe-1572-4857-8931-19159b9b4280.png">

"I've encountered the same problem. Have you already resolved it?"

**Comment by BrianG13** [comment]
Same problem here.
Any news?

**Comment by 478786359** [comment]
> Same problem here. Any news?

This is because I use multiple graphics cards when running. If I only use one, this problem will be solved.

**Comment by haotian-liu** [comment]
This may have also been fixed in #720. Please feel free to re-open if the problem persists. Thanks.

**Comment by zhanghuiecho** [comment]
> This may have also been fixed in [#720](https://github.com/haotian-liu/LLaVA/pull/720). Please feel free to re-open if the problem persists. Thanks.

I also meet this error, when i finetune llava1.5 in my data, and i wirte a test inference code based cli.py, just repeat load the whole finetuned model to process every text-image pair, but this error just happen in some data, all of my test data is 93, there are 5 data encounter this error ,i don't know how to solved it ,  :(  :(  :(  :( my code and error as follow，could you help me, thank you :) ：），and i also have some other questions ,i want to finetune this model only  text modal and image modal, how to do this :
`import os
os.environ["CUDA_VISIBLE_DEVICES"] = "0" 
import argparse
import torch
import json
from llava.constants import IMAGE_TOKEN_INDEX, DEFAULT_IMAGE_TOKEN, DEFAULT_IM_START_TOKEN, DEFAULT_IM_END_TOKEN
from llava.conversation import conv_templates, SeparatorStyle
from llava.model.builder import load_pretrained_model
from llava.utils import disable_torch_init
from llava.mm_utils import process_images, tokenizer_image_token, get_model_name_from_path

from PIL import Image

import requests
from PIL import Image
from io import BytesIO
from transformers import TextStreamer


def load_image(image_file):
    if image_file.startswith('http://') or image_file.startswith('https://'):
        response = requests.get(image_file)
        image = Image.open(BytesIO(response.content)).convert('RGB')
    else:
        image = Image.open(image_file).convert('RGB')
    return image


def test(image_file,input_text):
    # Model
    disable_torch_init()

    model_name = get_model_name_from_path(args.model_path)
    tokenizer, model, image_processor, context_len = load_pretrained_model(args.model_path, args.model_base, model_name,
                                                                           args.load_8bit, args.load_4bit,
                                                                           device=args.device)
    if "llama-2" in model_name.lower():
        conv_mode = "llava_llama_2"
    elif "mistral" in model_name.lower():
        conv_mode = "mistral_instruct"
    elif "v1.6-34b" in model_name.lower():
        conv_mode = "chatml_direct"
    elif "v1" in model_name.lower():
        conv_mode = "llava_v1"
    elif "mpt" in model_name.lower():
        conv_mode = "mpt"
    else:
        conv_mode = "llava_v0"

    if args.conv_mode is not None and conv_mode != args.conv_mode:
        print(
            '[WARNING] the auto inferred conversation mode is {}, while `--conv-mode` is {}, using {}'.format(conv_mode,
                                                                                                              args.conv_mode,
                                                                                                              args.conv_mode))
    else:
        args.conv_mode = conv_mode

    conv = conv_templates[args.conv_mode].copy()
    if "mpt" in model_name.lower():
        roles = ('user', 'assistant')
    else:
        roles = conv.roles

    image = load_image(image_file)
    image_size = image.size
    # Similar operation in model_worker.py
    image_tensor = process_images([image], image_processor, model.config)
    if type(image_tensor) is list:
        image_tensor = [image.to(model.device, dtype=torch.float16) for image in image_tensor]
    else:
        image_tensor = image_tensor.to(model.device, dtype=torch.float16)

    # while True:
    #     try:
    #         inp = input(f"{roles[0]}: ")
    #     except EOFError:
    #         inp = ""
    #     if not inp:
    #         print("exit...")
    #         break
    inp = input_text
    print(f"{roles[1]}: ", end="")

    if image is not None:
        # first message
        if model.config.mm_use_im_start_end:
            inp = DEFAULT_IM_START_TOKEN + DEFAULT_IMAGE_TOKEN + DEFAULT_IM_END_TOKEN + '\n' + inp
        else:
            inp = DEFAULT_IMAGE_TOKEN + '\n' + inp
        image = None

    conv.append_message(conv.roles[0], inp)
    conv.append_message(conv.roles[1], None)
    prompt = conv.get_prompt()

    input_ids = tokenizer_image_token(prompt, tokenizer, IMAGE_TOKEN_INDEX, return_tensors='pt').unsqueeze(0).to(model.device)
    stop_str = conv.sep if conv.sep_style != SeparatorStyle.TWO else conv.sep2
    keywords = [stop_str]
    streamer = TextStreamer(tokenizer, skip_prompt=True, skip_special_tokens=True)

    with torch.inference_mode():
        output_ids = model.generate(
            input_ids,
            images=image_tensor,
            image_sizes=[image_size],
            do_sample=True if args.temperature > 0 else False,
            temperature=args.temperature,
            max_new_tokens=args.max_new_tokens,
            streamer=streamer,
            use_cache=True)

    outputs = tokenizer.decode(output_ids[0]).strip()
    conv.messages[-1][-1] = outputs

    if args.debug:
        # print("\n", {"prompt": prompt, "outputs": outputs}, "\n")
        print("\n",  "outputs", outputs, "\n")
    return outputs


def main(args):
    #test data
    with open(args.test_file, "r", encoding="utf-8") as f:
        all_test = json.load(f)
    f.close()

    all_pred_all = []
    with open("llava_pred.json", "w", encoding="utf-8") as f1:
        for i, test_data_i in enumerate(all_test[69:]):
            all_pred = {}
            id, image_i, text_i = test_data_i['id'], test_data_i['image'], test_data_i['conversations']

            image_i_new = r'/root/image/image/' + image_i
            text_i_new = text_i[0]['value'].split('image>\n')[1]

            all_pred['id'] = i
            all_pred['text_id'] = id
            all_pred['image_id'] = image_i
            all_pred['gold'] = text_i[1]['value']
            print(image_i_new)
            print(id)
            print(f'gold: {i+69}',all_pred['gold'])
            output_i = test(image_i_new, text_i_new )
            all_pred['pred'] = output_i
            all_pred_all.append(all_pred)

            json.dump(all_pred, f1, indent=4, sort_keys=False, ensure_ascii=False)
            f1.write("\n")
    f1.close()

    with open("llava_pred.json", "w", encoding="utf-8") as f:
        json.dump(all_pred_all, f, indent=4, sort_keys=False)


if __name__ == "__main__":
    parser = argparse.ArgumentParser()

    parser.add_argument("--model-path", type=str, default="../../checkpoints/llava-v1.5-7b-task-lora-merged_mmee")
    parser.add_argument("--model-base", type=str, default=None)
    parser.add_argument("--image-file", type=str, default="xxx.jpg")
    parser.add_argument("--test-file", type=str, default="/root/LLaVA/playground/data/test/llava_mmee_test.json")
    parser.add_argument("--device", type=str, default="cuda")
    parser.add_argument("--conv-mode", type=str, default=None)
    parser.add_argument("--temperature", type=float, default=0.2)
    parser.add_argument("--max-new-tokens", type=int, default=512)
    parser.add_argument("--load-8bit", action="store_true")
    parser.add_argument("--load-4bit", action="store_true")
    parser.add_argument("--debug", action="store_true")
    args = parser.parse_args()
    main(args)


"""
 File "/root/autodl-tmp/LLaVA/llava/serve/cli_pred.py", line 170, in <module>
    args = parser.parse_args()
  File "/root/autodl-tmp/LLaVA/llava/serve/cli_pred.py", line 143, in main
    output_i = test(image_i_new, text_i_new )
  File "/root/autodl-tmp/LLaVA/llava/serve/cli_pred.py", line 102, in test
    output_ids = model.generate(
  File "/root/miniconda3/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/root/autodl-tmp/LLaVA/llava/model/language_model/llava_llama.py", line 137, in generate
    return super().generate(
  File "/root/miniconda3/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/root/miniconda3/lib/python3.10/site-packages/transformers/generation/utils.py", line 1525, in generate
    return self.sample(
  File "/root/miniconda3/lib/python3.10/site-packages/transformers/generation/utils.py", line 2658, in sample
    next_tokens = torch.multinomial(probs, num_samples=1).squeeze(1)
**RuntimeError: probability tensor contains either `inf`, `nan` or element < 0**

Process finished with exit code 1

"""
`
